src/v/kafka/server/tests/alter_user_scram_credentials_test.cc

```diff
@@ -609,3 +609,329 @@ FIXTURE_TEST(
+            .salt = creds_512.salt(),
+            .salted_password = salted_password_512,
+          });
+
+        auto resp
+          = client.dispatch(std::move(req), kafka::api_version(0)).get();
+        BOOST_REQUIRE(resp.data.errored());
+        BOOST_REQUIRE_EQUAL(resp.data.results.size(), 1);
+        BOOST_CHECK_EQUAL(resp.data.results[0].user, user_name_256);
+        BOOST_CHECK_EQUAL(
+          resp.data.results[0].error_code,
+          kafka::error_code::cluster_authorization_failed);
+
+        auto& sec = app.controller->get_credential_store().local();
+        BOOST_REQUIRE(sec.contains(security::credential_user(user_name_256)));
+        auto cred = sec.get<security::scram_credential>(
+          security::credential_user(user_name_256));
+        BOOST_REQUIRE(cred.has_value());
+        BOOST_CHECK_EQUAL(cred, creds_256);
+    }
+
+    // a user cannot delete a superuser
+    {
+        kafka::alter_user_scram_credentials_request req;
+        req.data.deletions.emplace_back(
+          kafka::scram_credential_deletion{
+            .name = kafka::scram_user_name{user_name_256},
+            .mechanism = kafka::scram_mechanism::scram_sha_256});
+
+        auto resp
+          = client.dispatch(std::move(req), kafka::api_version(0)).get();
+        BOOST_REQUIRE(resp.data.errored());
+        BOOST_REQUIRE_EQUAL(resp.data.results.size(), 1);
+        BOOST_CHECK_EQUAL(resp.data.results[0].user, user_name_256);
+        BOOST_CHECK_EQUAL(
+          resp.data.results[0].error_code,
+          kafka::error_code::cluster_authorization_failed);
+
+        auto& sec = app.controller->get_credential_store().local();
+        BOOST_REQUIRE(sec.contains(security::credential_user(user_name_256)));
+    }
+}
+
+FIXTURE_TEST(
+  alter_user_scram_credentials_user_to_superuser_noauth,
+  alter_user_scram_credentials_fixture) {
+    wait_for_controller_leadership().get();
+
+    ss::smp::invoke_on_all([]() {
+        config::shard_local_cfg()
+          .get("superusers")
+          .set_value(std::vector<ss::sstring>{user_name_256});
+    }).get();
+    auto deferred_config = ss::defer([] {
+        ss::smp::invoke_on_all([]() {
+            config::shard_local_cfg()
+              .get("superusers")
+              .set_value(std::vector<ss::sstring>{});
+        }).get();
+    });
+
+    auto creds_256 = security::scram_sha256::make_credentials(
+      password_256, security::scram_sha256::min_iterations);
+    create_user(user_name_256, creds_256);
+
+    auto [creds_512, salted_password_512]
+      = security::scram_sha512::make_credentials_and_return_password(
+        password_512, security::scram_sha512::min_iterations);
+    create_user(user_name_512, creds_512);
+
+    auto client = make_kafka_client().get();
+    auto deferred_close = ss::defer([&client] { client.stop().get(); });
+    client.connect().get();
+
+    // an user cannot update a superuser even when auth checks are disabled
```

Comment:
This is AuthZ, right? Please clarify in the comment.

If authorization is not required anywhere in the cluster, why would we not want to be able to update superusers credentials?

It can be done with admin API, right?